### PR TITLE
gateway: serverAddr must be local and have a port

### PR DESCRIPTION
**Pull request by krisis**
## Description

This change validates serverAddr command line argument to make the failure message independent of the runtime.

## Motivation and Context

See #4503
## How Has This Been Tested?

Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by balamurugana — APPROVED** [review]

**Review by balamurugana — APPROVED** [review]
Please fix travis-ci failure.

**Comment by krisis** [comment]
@bala, from initial investigation of the travis-ci it appears that `TestInitListeners` is bound to fail unpredictably. We use `getFreePort` to find a port which is available and assume it would remain unused during the test. This is incorrect.

In summary, the CI failure is not related to this PR.

File: cmd/gateway-main.go [hunk 1]
```diff
@@ -209,6 +209,7 @@ func gatewayMain(ctx *cli.Context) {
 	endpointAddr := ctx.Args().Get(1)
 	// Third argument is the address flag
 	serverAddr := ctx.String("address")
+	fatalIf(CheckLocalServerAddr(serverAddr), "Invalid address ‘%s’ in command line argument.", serverAddr)
```
Comment (harshavardhana) [inline cmd/gateway-main.go:212]: This is fixed in the GCS branch . Can you check @krisis

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/gateway-main.go (hunk 1 above)
Comment (krisis) [reply cmd/gateway-main.go:212]: @harshavardhana, are we merging GCS branch onto master before the next release? If so I can close this PR.

Comment (harshavardhana) [reply cmd/gateway-main.go:212]: The better question to ask is is this a show stopper fix for this release. GCS is the next after this one..so  it doesn't make sense to duplicate a fix. If needed IMO the original fix sent to GCS should be sent here by the same submitter.

Comment (krisis) [reply cmd/gateway-main.go:212]: @harshavardhana, I don't understand why the original submitter alone should fix this. Is there any specific reason for this policy?

Are we only fixing show stopper issues now? This fix is simple and it's easy to understand its impact. I don't see a reason to wait for another release. Further, the point about duplicate fix is weak for a one line fix. Close the PR if you still feel strongly.

**Comment by harshavardhana** [comment]
@vadmeste can you take a look at this change?

**Comment by harshavardhana** [comment]
> @harshavardhana, I don't understand why the original submitter alone should fix this. Is there any specific reason for this policy?
> 

It is to avoid causing history and choosing commits. In such a scenario this commit would become dummy and lost in git history. If the original fix was sent here then its better to rebase and handle.  I leave it upto you. This PR is fine as is my only opinion was about keeping the history to avoid two commits addressing the same problem.

> Are we only fixing show stopper issues now? This fix is simple and it's easy to understand its impact. I don't see a reason to wait for another release. Further, the point about duplicate fix is weak for a one line fix. Close the PR if you still feel strongly.

Generally yes show stopper issues, but since its a UI issue it is a show stopper.

**Comment by krisis** [comment]
Closing this PR in favour of https://github.com/minio/minio/pull/4376

**Pull request closed without merging**